src/main/java/com/aws/greengrass/shadowmanager/sync/CloudDataClient.java

```diff
@@ -145,74 +120,78 @@ public synchronized void updateSubscriptions(Set<Pair<String, String>> shadowSet
-        deleteTopicsToRemove.removeAll(deleteTopics);
+            Set<String> deleteTopicsToRemove = new HashSet<>(subscribedDeleteShadowTopics);
+            deleteTopicsToRemove.removeAll(deleteTopics);
 
-        Set<String> deleteTopicsToSubscribe = new HashSet<>(deleteTopics);
-        deleteTopicsToSubscribe.removeAll(subscribedDeleteShadowTopics);
+            Set<String> deleteTopicsToSubscribe = new HashSet<>(deleteTopics);
+            deleteTopicsToSubscribe.removeAll(subscribedDeleteShadowTopics);
 
-        boolean success;
-        try {
-            success = RetryUtils.runWithRetry(RETRY_CONFIG, () -> {
-                unsubscribeToShadows(subscribedUpdateShadowTopics, updateTopicsToRemove, this::handleUpdate);
-                subscribeToShadows(subscribedUpdateShadowTopics, updateTopicsToSubscribe, this::handleUpdate);
+            boolean success;
+            try {
+                success = RetryUtils.runWithRetry(RETRY_CONFIG, () -> {
+                    unsubscribeToShadows(subscribedUpdateShadowTopics, updateTopicsToRemove, this::handleUpdate);
+                    subscribeToShadows(subscribedUpdateShadowTopics, updateTopicsToSubscribe, this::handleUpdate);
 
-                unsubscribeToShadows(subscribedDeleteShadowTopics, deleteTopicsToRemove, this::handleDelete);
-                subscribeToShadows(subscribedDeleteShadowTopics, deleteTopicsToSubscribe, this::handleDelete);
+                    unsubscribeToShadows(subscribedDeleteShadowTopics, deleteTopicsToRemove, this::handleDelete);
+                    subscribeToShadows(subscribedDeleteShadowTopics, deleteTopicsToSubscribe, this::handleDelete);
 
-                if (!updateTopicsToRemove.isEmpty() || !updateTopicsToSubscribe.isEmpty()
-                        || !deleteTopicsToRemove.isEmpty() || !deleteTopicsToSubscribe.isEmpty()
-                        && !Thread.currentThread().isInterrupted()) {
+                    if (!updateTopicsToRemove.isEmpty() || !updateTopicsToSubscribe.isEmpty()
+                            || !deleteTopicsToRemove.isEmpty() || !deleteTopicsToSubscribe.isEmpty()
+                            && !Thread.currentThread().isInterrupted()) {
 
-                    throw new SubscriptionRetryException("Missed shadow topics to (un)subscribe to");
-                }
+                        throw new SubscriptionRetryException("Missed shadow topics to (un)subscribe to");
+                    }
 
-                // if interrupted then handle
-                if (Thread.currentThread().isInterrupted()) {
-                    Thread.currentThread().interrupt();
-                    logger.atError()
-                            .setEventType(LogEvents.CLOUD_DATA_CLIENT_SUBSCRIPTION_ERROR.code())
-                            .log("Failed to update subscriptions");
-                    return false;
-                }
+                    // if interrupted then handle
+                    if (Thread.currentThread().isInterrupted()) {
```

Comment:
i realize this is pre-existing, but let's just get rid of this whole block.  the subscribe/unsubscribe methods above will throw InterruptedException.  And returning `false` here on interrupt is wrong imo, the operation *did* succeed at this point.   Let's also adjust the logs while we're at it:  interruption isn't an error, it's part of the expected behavior/flow, either shutdown or we're doing a new task, so "failed to update" isn't true